echidna/limit/fit.py

```diff
@@ -0,0 +1,489 @@
+                                    spectra.get_config().get_par(par)._low))
+            if not numpy.isclose(self._roi[dim][1],
+                                 spectra.get_config().get_par(par)._high):
+                raise ValueError("roi %s high (%s) not equal to spectra %s"
+                                 " %s high (%s)"
+                                 % (dim, self.roi[dim][0], dim,
+                                    spectra.get_config().get_par(par)._high))
+
+    def get_data(self):
+        """ Gets the data you are fitting.
+
+        Returns:
+          :class:`echidna.core.spectra.Spectra`: The data you are fitting.
+        """
+        return self._data
+
+    def get_fixed_background(self):
+        """ Gets the fixed background you are fitting.
+
+        Returns:
+          :class:`echidna.core.spectra.Spectra`: The fixed background you are
+            fitting.
+        """
+        return self._fixed_background
+
+    def get_floating_backgrounds(self):
+        """ Gets the floating backgrounds you are fitting.
+
+        Returns:
+          list: The floating backgrounds you are fitting.
+        """
+        return self._floating_backgrounds
+
+    def get_method(self):
+        """ Gets the method you are using to calculate the test statistic you
+          are using to fit.
+
+        Returns:
+          :class:`TBC`: The method used to calculate test statistics.
+        """
+        return self._method
+
+    def get_roi(self):
+        """ Gets the region of interest (roi)
+
+        Returns:
+          dict: The region of interest
+        """
+        return self._roi
+
+    def get_roi_pars(self, spectra):
+        """ Get the parameters of a spectra that contain the roi.
+
+        Args:
+          :class:`echidna.core.spectra.Spectra`: The spectra you want to obtain
+            the roi parameters for.
+
+        Returns:
+          list: Of the names of the spectra parameters which contain the roi.
+        """
+        pars = []
+        for dim in self._roi:
+            dim_type = spectra.get_config().get_dim_type(dim)
+            par = dim + "_" + dim_type
+            pars.append(par)
+        return pars
+
+    def get_signal(self):
+        """ Gets the signal you are fitting.
+
+        Returns:
+          :class:`echidna.core.spectra.Spectra`: The fixed background you are
+            fitting.
+        """
+        return self._signal
+
+    def get_statistic(self):
+        """ Gets the value of the test statistic used for fitting.
+
+        Returns:
+          float or :class:`numpy.array`: The resulting test statisic(s)
+            dependent upon what method is used to compute the statistic.
+        """
+        if not self._checked:
+            self.check_fitter()
+        if not self._floating_backgrounds:
+            observed = self._data.nd_project(self._data_pars)
+            expected = self._fixed_background.nd_project(self._fixed_pars)
```

Comment:
@jwaterfield, quick question, why do you pass `_data_pars` and `_fixed_pars` here, instead of `get_spectra_config.get_pars`?